### PR TITLE
Remove duplicated code in Threads::ParallelScan_Range

### PR DESCRIPTION
**Pull request by Rombur**
This PR just moves duplicated code to two free functions.

### CODE REVIEW

**Review by masterleinad — COMMENTED** [review]
This still duplicates a majority of the implementation for the two classes. For other backends, we derive the two classes from a `ParallelScan*Base`, see https://github.com/kokkos/kokkos/blob/91ee4e1e145b4daf69c6457ab35c7be0a4cda9f2/core/src/SYCL/Kokkos_SYCL_ParallelScan_Range.hpp#L362-L413.

File: core/src/Threads/Kokkos_Threads_ParallelScan_Range.hpp [hunk 1]
```diff
@@ -21,6 +21,34 @@
 
 namespace Kokkos {
 namespace Impl {
+namespace Details {
```
Comment (ldh4) [inline core/src/Threads/Kokkos_Threads_ParallelScan_Range.hpp:24]: Is it necessary to introduce this namespace? Is `Impl` namespace not enough?

Comment (crtrott) [reply core/src/Threads/Kokkos_Threads_ParallelScan_Range.hpp:24]: Yeah we don't really need more or?

**Comment by dalg24** [comment]
What is the status of this PR?

**Comment by dalg24** [comment]
HIP runtime failures are unrelated (machine issue)

**Comment by Rombur** [comment]
> What is the status of this PR?

It's on my TODO list to finish. There are things more important than this PR that I want to finish.

**Comment by Rombur** [comment]
This is too outdated. Closing.

**Pull request closed without merging**